### PR TITLE
Prove Delete Lemmas for ReplicaSet in V2 State Machine

### PR DESCRIPTION
**Pull request by codyjrivera**
Prove delete WF1 lemmas for ReplicaSet in V2 state machine.

### CODE REVIEW

File: src/v2/controllers/vreplicaset_controller/proof/helper_invariants/predicate.rs [hunk 1]
```diff
@@ -198,7 +198,9 @@ pub open spec fn every_delete_matching_pod_request_implies_at_after_delete_pod_s
     |s: ClusterState| {
         forall |msg: Message| #![trigger s.in_flight().contains(msg)] {
             let content = msg.content;
-            let key = content.get_delete_request().key;
+            let req = content.get_delete_request();
+            let req_rv = req.preconditions.unwrap().resource_version.unwrap();
+            let key = req.key;
```
Comment (marshtompsxd) [inline src/v2/controllers/vreplicaset_controller/proof/helper_invariants/predicate.rs:203]: nit: change 201-203 does not seem necessary?

File: src/v2/controllers/vreplicaset_controller/proof/helper_invariants/predicate.rs [hunk 2]
```diff
@@ -209,42 +211,28 @@ pub open spec fn every_delete_matching_pod_request_implies_at_after_delete_pod_s
...
+            let obj = s.resources()[key];
             &&& exists |diff: usize| #[trigger] at_vrs_step_with_vrs(vrs, controller_id, VReplicaSetReconcileStep::AfterDeletePod(diff))(s)
             &&& Cluster::pending_req_msg_is(controller_id, s, vrs.object_ref(), msg)
+
+            // Delete precondition clauses.
+            &&& req.preconditions.is_Some()
+            &&& req.preconditions.unwrap().resource_version.is_Some()
+            &&& obj.metadata.uid.is_None()
+            &&& obj.metadata.resource_version.is_Some()
+            &&& obj.metadata.resource_version.unwrap() == req_rv
```
Comment (marshtompsxd) [inline src/v2/controllers/vreplicaset_controller/proof/helper_invariants/predicate.rs:227]: @codyjrivera could you add a comment here saying that we require the RV version carried by the Delete to match the RV of the stored object to exclude the case that other reconcile procedures (working on other VRS objects) of the VRS controller try to delete the same object.

File: src/v2/controllers/vreplicaset_controller/proof/helper_invariants/predicate.rs [hunk 3]
```diff
@@ -209,42 +211,28 @@ pub open spec fn every_delete_matching_pod_request_implies_at_after_delete_pod_s
...
             &&& owned_selector_match_is(vrs, obj)
         } ==> {
+            let content = msg.content;
+            let req = content.get_delete_request();
+            let req_rv = req.preconditions.unwrap().resource_version.unwrap();
+            let key = req.key;
+            let obj = s.resources()[key];
             &&& exists |diff: usize| #[trigger] at_vrs_step_with_vrs(vrs, controller_id, VReplicaSetReconcileStep::AfterDeletePod(diff))(s)
             &&& Cluster::pending_req_msg_is(controller_id, s, vrs.object_ref(), msg)
+
```
Comment (marshtompsxd) [inline src/v2/controllers/vreplicaset_controller/proof/helper_invariants/predicate.rs:221]: nit: delete this blank line

File: src/v2/controllers/vreplicaset_controller/proof/helper_invariants/predicate.rs (hunk 1 above)
Comment (codyjrivera) [reply src/v2/controllers/vreplicaset_controller/proof/helper_invariants/predicate.rs:203]: Eliminating some of them, kept others.

File: src/v2/controllers/vreplicaset_controller/proof/helper_invariants/predicate.rs (hunk 2 above)
Comment (codyjrivera) [reply src/v2/controllers/vreplicaset_controller/proof/helper_invariants/predicate.rs:227]: Will specify this.

**Pull request merged**